src/collection.ts

```diff
@@ -1582,7 +1647,7 @@ export class Collection<TSchema extends Document = Document> {
    * one will be added to each of the documents missing it by the driver, mutating the document. This behavior
    * can be overridden by setting the **forceServerObjectId** flag.
    *
-   * @deprecated Use insertOne, insertMany or bulkWrite instead.
+   * @deprecated Use insertOne, insertMany or bulkWrite instead. Callbacks are deprecated and will be removed in the next major version
```

Comment:
This comment applies to any previously deprecated functions that take callbacks in this PR.

How do want to handle these cases?  I couldn't find an existing ticket, so as far as I can tell we have no plans to remove these functions in 5.0.  My preference would be to remove these functions, and if so, I think we should 1. file a jira ticket for their removal and make sure it's tagged for 5.0 and 2. remove the callback specific deprecation.

If we don't decide to go this route - I think we should properly overload all these functions and deprecate the callback overloads specifically, mentioning their removal in v5.  

Thoughts?

Reply:
https://jira.mongodb.org/browse/NODE-4614

Reply:
Can we remove the callback deprecation warning here as well?

Reply:
We should mention both but maybe I could rephrase? since currently migrating off of `.insert` is 
```diff
- c.insert({ doc: 1 }, callback)
+ c.insertMany([{ doc: 1 }], callback)
```
Now that callbacks will be removed in addition it is more involved than just wrapping the first argument in brackets

Reply:
I still think we should only mention the whole deprecation.  `insert` (and similar functions) will be entirely removed in v5.  I think the user flow of 

- user sees insert is deprecated and switches to insertOne 
- user sees insertOne with callbacks is deprecated and switches to promises 

is fine because at each stage, they see an specific deprecation warning and know how to resolve it.  a single warning that combines both (i.e., `Use insertOne, insertMany or bulkWrite instead. Callbacks are deprecated and will be removed in the next major version`) is more confusing.  Maybe we should ask the team / Matt

Reply:
@dariakp @durran

Reply:
from the conversation in slack, callback deprecation and removal is distinct and more certain than the whole api deprecation and removal, we should have separate notes so that the corresponding tasks can be completed separately